custom_components/localtuya/cover.py

```diff
@@ -49,33 +54,49 @@
     CONF_OPEN_CMD,
     CONF_CLOSE_CMD,
     CONF_STOP_CMD,
+    CONF_GET_POSITION_DPS,
+    CONF_SET_POSITION_DPS,
+    CONF_LAST_MOVEMENT_DPS
 )
-from .const import CONF_OPEN_CMD, CONF_CLOSE_CMD, CONF_STOP_CMD
+
 from .pytuya import TuyaDevice
 
 _LOGGER = logging.getLogger(__name__)
 
-DEFAULT_OPEN_CMD = "on"
-DEFAULT_CLOSE_CMD = "off"
+DEFAULT_OPEN_CMD = "open"
+DEFAULT_CLOSE_CMD = "close"
 DEFAULT_STOP_CMD = "stop"
+DEFAULT_SET_POSITION_DPS = 0
+DEFAULT_GET_POSITION_DPS = 0
+DEFAULT_LAST_MOVEMENT_DPS = 0
+
+MIN_POSITION = 0
+MAX_POSITION = 100
+UPDATE_RETRY_LIMIT = 3
 
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(BASE_PLATFORM_SCHEMA).extend(
     {
         vol.Optional(CONF_OPEN_CMD, default=DEFAULT_OPEN_CMD): cv.string,
         vol.Optional(CONF_CLOSE_CMD, default=DEFAULT_CLOSE_CMD): cv.string,
         vol.Optional(CONF_STOP_CMD, default=DEFAULT_STOP_CMD): cv.string,
+        vol.Optional(CONF_SET_POSITION_DPS, default=DEFAULT_SET_POSITION_DPS): cv.positive_int, # TODO only allow user to select from returned dps list
+        vol.Optional(CONF_GET_POSITION_DPS, default=DEFAULT_GET_POSITION_DPS): cv.positive_int, # TODO only allow user to select from returned dps list
+        vol.Optional(CONF_LAST_MOVEMENT_DPS, default=DEFAULT_LAST_MOVEMENT_DPS): cv.positive_int, # TODO only allow user to select from returned dps list
     }
 )
 
 
 def flow_schema(dps):
     """Return schema used in config flow."""
     return {
-        vol.Optional(CONF_OPEN_CMD, default=DEFAULT_OPEN_CMD): str,
-        vol.Optional(CONF_CLOSE_CMD, default=DEFAULT_CLOSE_CMD): str,
-        vol.Optional(CONF_STOP_CMD, default=DEFAULT_STOP_CMD): str,
-    }
+            vol.Optional(CONF_OPEN_CMD, default=DEFAULT_OPEN_CMD): str,
+            vol.Optional(CONF_CLOSE_CMD, default=DEFAULT_CLOSE_CMD): str,
+            vol.Optional(CONF_STOP_CMD, default=DEFAULT_STOP_CMD): str,
+            vol.Optional(CONF_SET_POSITION_DPS, default=DEFAULT_SET_POSITION_DPS): cv.positive_int,
```

Comment:
Did you verify that this worked? I've had problems using `cv.string` with config flows, doesn't seem to be supported. Is it ok with `cv.positive_int`? I got an exception.

Reply:
I have had no problem with `str` or `cv.positive_int`.

Reply:
Great! 👍